server/render/index.js

```diff
@@ -97,7 +97,12 @@ export function serverRender( req, res ) {
 
 		let reduxSubtrees = [ 'documentHead' ];
 		if ( isSectionIsomorphic( context.store.getState() ) ) {
-			reduxSubtrees = reduxSubtrees.concat( [ 'ui', 'themes' ] );
+			reduxSubtrees = reduxSubtrees.concat( [ 'ui' ] );
+
+			// Send themes redux state only in logged-out scenario
+			if ( ! context.user ) {
```

Comment:
We can put _all_ of the redux state behind this check to make things simpler. I don't think there is a compelling reason for sending only the `ui` tree.